src/openassetio-core/include/openassetio/utils/path.hpp

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 The Foundry Visionmongers Ltd
+#pragma once
+#include <cstdint>
+#include <memory>
+
+#include <openassetio/export.h>
+#include <openassetio/typedefs.hpp>
+
+namespace openassetio {
+inline namespace OPENASSETIO_CORE_ABI_VERSION {
+namespace utils {
+
+/**
+ * Indicator of platform type associate with a path.
+ */
+enum class PathType : std::uint8_t {
+  /// Use the current system platform to determine path type.
+  kSystem = 0,
+  /// Assume a POSIX path.
+  kPOSIX,
+  /// Assume a Windows path (including UNC).
+  kWindows
+};
+
+/**
+ * Construct a file URL from a path.
+ *
+ * When processing many paths, where performance is a concern, construct
+ * construct a @ref FileUrlPathConverter instead.
+ *
+ * @param absolutePath Path string.
+ * @param pathType Platform associated with path.
+ * @return Converted file URL.
+ * @throws InputValidationException if the path is invalid or
+ * unsupported.
+ */
+OPENASSETIO_CORE_EXPORT Str pathToUrl(StrView absolutePath, PathType pathType = PathType::kSystem);
+
+/**
+ * Convert a file URL to a path.
+ *
+ * When processing many URLs, where performance is a concern, construct
+ * construct a @ref FileUrlPathConverter instead.
```

Comment:
[Consider][strong] Regardless of which way we go, do you think exposing duplicate public interfaces for this is a good idea? There needs to be a pretty strong reason in my head to introduce that decision point on a user.

Reply:
Related to https://github.com/OpenAssetIO/OpenAssetIO/pull/1246#discussion_r1461581901 - although I guess this covers whether we should have _both_ interfaces.

If we do choose just one, I'd rather expose the class than the free functions, for reasons examined in the linked thread.  I don't have a strong objection to removing the free functions (ACs be damned).  Thoughts?

Reply:
If we do end up keeping the class then yes, I would also remove the free functions. I believe this makes the interface worse, but it's better than having 2 parallel implementations.

Reply:
Removed free functions in 9d6af5d4